File: app/src/test/java/com/fractalwrench/crazycats/image/list/ListPresenterSuccessTest.java
package com.fractalwrench.crazycats.image.list;


import com.fractalwrench.crazycats.mocks.MockImageListView;
import com.fractalwrench.crazycats.mocks.TestDependencies;

import org.junit.After;
import org.junit.Before;
import org.junit.Test;

import java.io.IOException;

import static org.junit.Assert.assertNotNull;
import static org.junit.Assert.assertNull;

public class ListPresenterSuccessTest {

    private ImageListPresenter presenter;
    private MockImageListView listView;

    @Before
    public void setup() throws IOException {
        presenter = new ImageListPresenter(TestDependencies.mockSuccessRepository());
        listView = new MockImageListView();
    }

    @After
    public void tearDown() {
        presenter.stop();
    }

    /**
     * Ensures that the view is notified when the presenter fetches data
     */
    @Test
    public void checkPostListSuccess() {
        assertNull(listView.content);
        assertNull(listView.errorMessage);

        presenter.start(listView);
        assertNotNull(listView.content);
        assertNull(listView.errorMessage);
    }

}

File: app/src/test/java/com/fractalwrench/crazycats/image/list/ListPresenterSuccessTest.kt
package com.fractalwrench.crazycats.image.list


import com.fractalwrench.crazycats.mocks.MockImageListView
import com.fractalwrench.crazycats.mocks.TestDependencies

import org.junit.After
import org.junit.Before
import org.junit.Test

import java.io.IOException

import org.junit.Assert.assertNotNull
import org.junit.Assert.assertNull

class ListPresenterSuccessTest {

    private var presenter: ImageListPresenter? = null
    private var listView: MockImageListView? = null

    @Before
    @Throws(IOException::class)
    fun setup() {
        presenter = ImageListPresenter(TestDependencies.mockSuccessRepository())
        listView = MockImageListView()
    }

    @After
    fun tearDown() {
        presenter!!.stop()
    }

    /**
     * Ensures that the view is notified when the presenter fetches data
     */
    @Test
    fun checkPostListSuccess() {
        assertNull(listView!!.content)
        assertNull(listView!!.errorMessage)

        presenter!!.start(listView!!)
        assertNotNull(listView!!.content)
        assertNull(listView!!.errorMessage)
    }

}
